Add information and example of combining SHOW DATABASES with regular Cypher (#2986)
neo4j/docs-operations: modules/ROOT/pages/database-administration/standard-databases/listing-databases.adoc

|===

> From 2026.XX, it is possible to combine the `SHOW DATABASES` command with regular Cypher syntax.

Author: ```suggestion
From 2026.05, it is possible to combine the `SHOW DATABASES` command with regular Cypher syntax.
```

> From 2026.05, it is possible to combine the `SHOW DATABASES` command with regular Cypher syntax.

Author: ```suggestion
In Cypher 25 from 2026.05, it is possible to combine the `SHOW DATABASES` command with regular Cypher syntax.
```

> However, since `SHOW DATABASES` still need to be run against the system database, all parts of the query must be allowed on the system database.

Reviewer: ```suggestion
However, since `SHOW DATABASES` is run against the `system` database, all parts of the query must be allowed on the `system` database.
```

> This means that clauses like `MATCH` or `SHOW INDEXES` are not allowed, as those are not allowed on the system database, but procedure calls of system-allowed procedures are allowed.

Reviewer: ```suggestion
For example, clauses like `MATCH` and `SHOW INDEXES` are not allowed on the `system` database, therefore, they cannot be used with `SHOW DATABASES`.
However, procedure calls of system-allowed procedures are allowed.
```

> If the `SHOW DATABASES` is combined with other clauses the automatic re-routing of the command to the system database is disabled and explicit connection or routing is required.

Reviewer 2: By explicit do we mean by using the Cypher USE clause? Or are there other ways of explicitly re-routing to system? If not, maybe we could refer to the USE clause here.

Author: in query routing I think is only the USE clause, but I also think we usually just say routing on these pages 🤷 and ``` explicit connection or `USE` clause is required``` sounds weirder to me

Reviewer 2: Yeah, if we usually just refer to routing then I agree it is better to be consistent and keep it that way.

Author: we have this page, https://neo4j.com/docs/operations-manual/current/database-administration/routing-decisions/, talking about routing so I don't think it's wrong to use routing at least

Reviewer 2: no, I agree 👍

> If the `SHOW DATABASES` is combined with other clauses the automatic re-routing of the command to the system database is disabled and explicit connection or routing is required.

Reviewer: ```suggestion
If the `SHOW DATABASES` is combined with other clauses, the automatic re-routing of the command to the `system` database is disabled, and explicit connection or routing is required.
```

+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
----

> From 2026.XX, if the given database is a parameter that evaluates to null the command will no longer fail on wrong type but instead return nothing.

Author: ```suggestion
From 2026.05, if the given database is a parameter that evaluates to null the command will no longer fail on wrong type but instead return nothing.
```

> From 2026.05, if the given database is a parameter that evaluates to null the command will no longer fail on wrong type but instead return nothing.

Author: ```suggestion
In Cypher 25 from 2026.05, if the given database is a parameter that evaluates to null, the command no longer fails on the wrong type but instead returns nothing.
```

> This is consistent with how nothing is returned when the given database name doesn't exist.

Reviewer: ```suggestion
This behavior aligns with not returning anything when the given database name does not exist.
```

| "system"  | "CYPHER 25"

2+d|Rows: 3
|===

> [role=label--new-2026.XX]

Author: ```suggestion
[role=label--new-2026.05]
```

> [role=label--new-2026.05]

Author: ```suggestion
[role=label--cypher-25 label--new-2026.05]
```

== Show databases with quarantined status and unquarantine them

The `SHOW DATABASES` command can be combined with procedure calls.

> Since it is not the only clause in the query it will not be automatically routed to the system database so we add the `USE` clause to ensure we're executing on there.

Reviewer: ```suggestion
To ensure the command is run against the `system` database, prepend it with the `USE` clause.
Combining `SHOW DATABASES` with other Cypher queries disables the automatic re-routing.
```

Author: ```suggestion
To ensure the command is run against the `system` database, prepend it with the `USE` clause.
Combining `SHOW DATABASES` with other Cypher clauses disables the automatic re-routing.
```

.Query
[source, cypher]
----
USE system
SHOW DATABASES
YIELD name, serverID, currentStatus
WHERE currentStatus='quarantined'
CALL dbms.unquarantineDatabase(serverID, name)
----

> Nothing is returned from the query as it ends with a procedure call of a procedure that doesn't return anything.

Reviewer: ```suggestion
Nothing is returned from the query, as it ends with a procedure call of a procedure that does not return anything.
```

| "system"  | "CYPHER 25"

2+d|Rows: 3
|===

[role=label--cypher-25 label--new-2026.05]

> == Show databases with quarantined status and unquarantine them

Reviewer 3: I had to read that sentence a few times before understanding. Is "unquarantine" the preferred terminology?
If not, how about:

```suggestion
== Show quarantined databases and restore access
```

Author: we have a lot of things called access already (read vs read write access to the database, the access privileges for connecting to the database) so I'm hesitant to say "restore access". 

The quarantine/unquarantine is more clusterings area than mine, but looking at the description of the procedure:
```
Lift quarantine from a database on a given server.
```

maybe saying
```suggestion
== Show quarantined databases and lift their quarantine
```
or
```suggestion
== Show quarantined databases and lift them out of quarantine
```

Reviewer 3: Makes sense. I like the first option! ("Show quarantined databases and lift their quarantine")

Author: Fixed in [002ae42](https://github.com/neo4j/docs-operations/pull/2986/commits/002ae426c9947fa64b7ef7cf559a4955b8d8d0b0)

The `SHOW DATABASES` command can be combined with procedure calls.
To ensure the command is run against the `system` database, prepend it with the `USE` clause.
Combining `SHOW DATABASES` with other Cypher clauses disables the automatic re-routing.

.Query
[source, cypher]
----
USE system
SHOW DATABASES
YIELD name, serverID, currentStatus
WHERE currentStatus='quarantined'

> CALL dbms.unquarantineDatabase(serverID, name)

Reviewer 4: The input arguments of the unquarantine procedure are `server` and `database`? Should we replace `(serverID, name)` with `(server, database)`?

Author: the `serverID` and `name` are the variable names (the output column names from show databases, you could do 
```
USE system
SHOW DATABASES
YIELD name AS database, serverID AS server, currentStatus
WHERE currentStatus='quarantined'
CALL dbms.unquarantineDatabase(server, database)
```
but I'm not sure if that is better 🤷

Reviewer 4: ok, if it unnecessarily complicates the query, let's keep it as it is.
A side question: When you run the unquarantine procedure, you can choose one of the optional arguments: 
`keepStateKeepStore`, `replaceStateKeepStore`, and `replaceStateReplaceStore`. Can you define one of them when combining with the the SHOW DB command?

Author: yeah, I just didn't add it as it was optional and I wouldn't re-use any of the output from show databases for it 🤷

Author: it doesn't make it more complicated, it's about the same so if you think it's clearer to match the input argument names of the procedure we can do that

Reviewer 4: then let's make it clearer 🙂

Author: updated the query and added a sentence about the operation argument in [9216222](https://github.com/neo4j/docs-operations/pull/2986/commits/9216222835a54008b0d676454970a4ffd17cbeb4)

----

> 

Reviewer 4: We have detailed information on how to use this procedure here - https://neo4j.com/docs/operations-manual/current/database-administration/standard-databases/errors/#quarantine.
Should we add a sentence:
For more information on the `dbms.unquarantineDatabase()` procedure, see xref:database-administration/standard-databases/errors.adoc#quarantine[Standard databases -> Error handling].

Author: Added in [9216222](https://github.com/neo4j/docs-operations/pull/2986/commits/9216222835a54008b0d676454970a4ffd17cbeb4)

> Nothing is returned from the query, as it ends with a procedure call of a procedure that does not return anything.

Reviewer 4: Does this mean that you in fact won't see which database (or databases) was/were quarantined and then released from this quarantine?

Author: Not unless you return something after the procedure call. But if you add a `RETURN name, serverID` at the end then you'd see the names and servers of the databases.

Author: Added a sentence about it in [9216222](https://github.com/neo4j/docs-operations/pull/2986/commits/9216222835a54008b0d676454970a4ffd17cbeb4)

> Nothing is returned from the query, as it ends with a procedure call of a procedure that does not return anything.

Author: Could do this I guess: I think we define void procedures on the CALL procedure page that just got moved 🤷 
```suggestion
Nothing is returned from the query, as it ends with a procedure call of a void procedure.
```
